lib/inspec/resources/sys_info.rb

```diff
@@ -3,117 +3,140 @@
-        else
-          inspec.powershell("$env:computername").stdout.chomp
-        end
+        @cache = WindowsInfo.new(inspec)
       else
-        skip_resource "The `sys_info.hostname` resource is not supported on your OS yet."
+        return skip_resource "The `sys_info` resource is not supported on your OS yet."
       end
     end
 
-    def linux_hostname(opt = nil)
-      if !opt.nil?
-        opt = case opt
-              when "f", "long", "fqdn", "full"
-                " -f"
-              when "d", "domain"
-                " -d"
-              when "i", "ip_address"
-                " -I"
-              when "s", "short"
-                " -s"
-              else
-                "ERROR"
-              end
-      end
-      if opt == "ERROR"
-        skip_resource "The `sys_info.hostname` resource is not supported with that option on your OS."
-      else
-        inspec.command("hostname#{opt}").stdout.chomp
-      end
+    def domain
+      @cache.info[:domain]
     end
 
-    def mac_hostname(opt = nil)
-      if !opt.nil?
-        opt = case opt
-              when "f", "long", "fqdn", "full"
-                " -f"
-              when "s", "short"
-                " -s"
-              else
-                "ERROR"
-              end
-      end
-      if opt == "ERROR"
-        skip_resource "The `sys_info.hostname` resource is not supported with that option on your OS."
-      else
-        inspec.command("hostname#{opt}").stdout.chomp
-      end
+    def ip_address
+      @cache.info[:ip]
     end
 
-    # returns the Manufacturer of the local system
     def manufacturer
-      os = inspec.os
-      if os.darwin?
-        "Apple Inc."
-      elsif os.linux?
-        inspec.command("cat /sys/class/dmi/id/sys_vendor").stdout.chomp
-      elsif os.windows?
-        inspec.powershell("Get-CimInstance -ClassName Win32_ComputerSystem | Select Manufacturer -ExpandProperty Manufacturer").stdout.chomp
-      else
-        skip_resource "The `sys_info.manufacturer` resource is not supported on your OS yet."
-      end
+      @cache.info[:manufacturer]
     end
 
-    # returns the ServerModel of the local system
     def model
-      os = inspec.os
-      if os.darwin?
-        inspec.command("sysctl -n hw.model").stdout.chomp
-      elsif os.linux?
-        inspec.command("cat /sys/class/dmi/id/product_name").stdout.chomp
-      elsif os.windows?
-        inspec.powershell("Get-CimInstance -ClassName Win32_ComputerSystem | Select Model -ExpandProperty Model").stdout.chomp
-      else
-        skip_resource "The `sys_info.model` resource is not supported on your OS yet."
-      end
+      @cache.info[:model]
+    end
+
+    def totalmemory_kb
+      @cache.info[:totalmemory_kb]
+    end
+
+    def hostname
+      @cache.info[:hostname]
+    end
+
+    def fqdn
+      @cache.info[:fqdn]
     end
 
     def to_s
       "System Information"
     end
   end
+
+  class LinuxInfo < SysInfo
```

Comment:
I see a LinuxInfo and a new WindowsInfo, but the existing functionality supports MacOS as well. We'll need a support class for Mac.